include/dd/Package.hpp

```diff
@@ -1675,17 +1687,27 @@ namespace dd {
                     edge[idx] = ResultEdge::zero;
                     for (auto k = 0U; k < rows; k++) {
                         LEdge e1{};
+                        auto  element1 = rows * i + k;
                         if (!x.isTerminal() && x.p->v == var) {
-                            e1 = x.p->e[rows * i + k];
+                            e1 = x.p->e[element1];
                         } else {
+                            // Effectively inserts an identity node
                             e1 = xCopy;
+                            if ((element1 == 1 or element1 == 2) and (std::is_same_v<LeftOperandNode, mNode> or std::is_same_v<LeftOperandNode, dNode>)) {
```

Comment:
Please do not use logical connectives like `and` or `or`. There are many instances of this leading to problems.
Prefer the standard notation of `&&` and `||`.